CMakeLists.txt

```diff
@@ -48,6 +48,7 @@ option(COVERALLS_UPLOAD "Package code coverage data to coveralls"       OFF)
 option(ON_TRAVIS        "Exclude special unit test on Travis CI"        OFF)
 option(WITH_C_API       "Compile PaddlePaddle with C-API(Prediction)"   OFF)
 option(WITH_GOLANG      "Compile PaddlePaddle with GOLANG"              OFF)
+option(GLIDE_INSTALL    "Download and install go dependencies "         ON)
```

Comment:
Sure, I'll add a design doc before going on working with this PR. 

I thought previous discussions in https://github.com/PaddlePaddle/Paddle/pull/2583 is the reason why I'm working on this.